src/Tools/ExternalAccess/Razor/Cohost/RazorDynamicRegistrationServiceFactory.cs

```diff
@@ -6,55 +6,84 @@
 using System.Composition;
 using System.Threading;
 using System.Threading.Tasks;
+using Microsoft.CodeAnalysis.Editor.Shared.Utilities;
 using Microsoft.CodeAnalysis.Host.Mef;
 using Microsoft.CodeAnalysis.LanguageServer;
 using Microsoft.CodeAnalysis.LanguageServer.Handler;
 using Newtonsoft.Json;
 using Roslyn.LanguageServer.Protocol;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.ExternalAccess.Razor.Cohost;
 
 [ExportCSharpVisualBasicLspServiceFactory(typeof(RazorDynamicRegistrationService), WellKnownLspServerKinds.AlwaysActiveVSLspServer), Shared]
 [method: ImportingConstructor]
 [method: Obsolete(MefConstruction.ImportingConstructorMessage, error: true)]
-internal sealed class RazorDynamicRegistrationServiceFactory([Import(AllowDefault = true)] IRazorCohostDynamicRegistrationService? dynamicRegistrationService) : ILspServiceFactory
+internal sealed class RazorDynamicRegistrationServiceFactory(
+    [Import(AllowDefault = true)] IUIContextActivationService? uIContextActivationService,
```

Comment:
Do we have to allow null here because this is used in vscode?

Reply:
This is null because I'm paranoid, and an interface defined in one layer, and implemented in another, seems like a recipe for null to me. This doesn't ship in VS Code, at least not in the `.roslyn` folder, but it is present in OOP, and who knows where else. It shouldn't be imported by anything other than the always active LSP server, so maybe this is worrying about nothing?

Reply:
Ah if it is in OOP it might not be present - but then editor features I don't think is in OOP, so maybe the interface definition needs to move down?  Not 100% sure on that one

Reply:
EditorFeatures is in OOP on my machine at least. Perhaps it shouldn't be, but in future if we split the EA into language server and editor features layers, this would stick with editors features anyway I think. Plus I can't see any obvious place to put this that makes sense in Features :P